src/custom/utils/logging/index.ts

```diff
@@ -27,3 +27,13 @@ export function constructSentryError(
 
   return { baseError, sentryError: constructedError, tags }
 }
+
+// checks response for non json/application return type and throw appropriate error
+export function checkAndThrowIfJsonSerialisableError(response: Response) {
+  // don't attempt json parse if not json response...
+  if (response.headers.get('Content-Type') !== 'application/json') {
+    throw new Error(
+      `Non JSON serialisable error detected with status code ${response.status}. ${JSON.stringify(response.statusText)}`
```

Comment:
I find this weird. 

1st, why do you stringify the statusText? ${JSON.stringify(response.statusText)}`, is not already a string? with OK, Not Found, etc? 

Also, wouldn't it be relevant the status code? 

Additionally, i find strange you mention `Non JSON serialisable error detected`. Errors don't need to be JSON, so i wouldn't remove that part of the message.  

Another thing i find strange is the function responsibility. It looks like you you create this to refactor between the 2 places where a similar logic is done. This is nice, but also the other logic seems very similar. Why not refactor the whole error handling and simplify the name and responsability: i.e. `handleErrorResponse(response)`

Reply:
made some changes but kept the separation between `_handleQuoteResponse` and `_handleOrderResponse` here [66d0bfe](https://github.com/gnosis/cowswap/pull/2531/commits/66d0bfe7cc5c2ba8f16ede2b64f8f5ee7d9378e9)

Reply:
@anxolin